client/app/services/custom-button-menu/custom-button-menu.component.js

```diff
@@ -0,0 +1,62 @@
+import './_custom-button-menu.sass'
+import templateUrl from './custom-button-menu.html'
+
+export const CustomButtonMenuComponent = {
+  bindings: {
+    customActions: '<',
+    serviceId: '<',
+    vmId: '<'
+  },
+  controller: CustomButtonMenuController,
+  controllerAs: 'vm',
+  templateUrl
+}
+
+/** @ngInject */
+function CustomButtonMenuController ($state, EventNotifications, CollectionsApi, RBAC) {
+  const vm = this
+  vm.hasRequiredRole = hasRequiredRole
+  vm.invokeCustomAction = invokeCustomAction
+  function hasRequiredRole (button) {
+    const acceptableRoles = button.visibility.roles
+
+    return RBAC.hasRole(...acceptableRoles)
+  }
+
+  function invokeCustomAction (button) {
+    if (button.enabled) {
+      if (button.resource_action && button.resource_action.dialog_id) {
+        const options = {
+          button: button,
+          serviceId: vm.serviceId
+        }
+        if (vm.vmId) {
+          options.vmId = vm.vmId
+        }
+        $state.go('services.custom_button_details', options)
+      } else if (vm.vmId) {
+        const data = { action: button.name }
+        CollectionsApi.post('vms', vm.vmId, {}, data)
+          .then(postSuccess)
+          .catch(postFailure)
+      } else {
+        const data = { action: button.name }
+        CollectionsApi.post('services', vm.serviceId, {}, data)
+          .then(postSuccess)
+          .catch(postFailure)
+      }
+    }
+
+    function postSuccess (response) {
+      if (response.success === false) {
+        EventNotifications.error(response.message)
+      } else {
+        EventNotifications.success(response.message)
+      }
+    }
+
+    function postFailure () {
```

Comment:
There's gonna be real valuable server feedback on the error here, [something similar to this](https://github.com/manageiq/manageiq-ui-service/blob/01c16e684a29a8aba449249b8028c79f223eead3/client/app/services/process-snapshots-modal/process-snapshots-modal.component.js) should be used here